mne/viz/topo.py

```diff
@@ -73,87 +76,335 @@ def iter_topography(info, layout=None, on_pick=None, fig=None,
+        for tick in ax.get_xaxis().get_major_ticks() + \
+                ax.get_yaxis().get_major_ticks():
+            tick.set_pad(2. * fontsize / 3.)
+            tick.label1 = tick._get_text1()
+
+    if plot_tick_labels is True:
+        # ax.set_xticklabels([('' if i % 2 else str(t))
+                            # for i, t in enumerate(xticks)])
+        ax.set_xticklabels(xticks)
+        ax.set_yticklabels(yticks)
+    else:
+        for tic in ax.xaxis.get_major_ticks():
+            tic.tick1On = tic.tick2On = False
         ax.set_xticklabels([])
+        for tic in ax.yaxis.get_major_ticks():
+            tic.tick1On = tic.tick2On = False
         ax.set_yticklabels([])
-        plt.setp(ax.get_xticklines(), visible=False)
-        plt.setp(ax.get_yticklines(), visible=False)
-        ch_idx = ch_names.index(name)
-        vars(ax)['_mne_ch_name'] = name
-        vars(ax)['_mne_ch_idx'] = ch_idx
-        vars(ax)['_mne_ax_face_color'] = axis_facecolor
-        yield ax, ch_idx
+
+    ax.xaxis.set_tick_params(width=linewidth, length=3, color=spine_color)
+    ax.yaxis.set_tick_params(width=linewidth, length=3, color=spine_color)
+
+    for item in ([ax.title, ax.xaxis.label, ax.yaxis.label] +
+                 ax.get_xticklabels() + ax.get_yticklabels()):
+        item.set_fontsize(fontsize)
+        item.set_color(spine_color)
+
+    [line.set_marker(mpll.TICKDOWN) for line in ax.get_xticklines()]
+    [line.set_marker(mpll.TICKLEFT) for line in ax.get_yticklines()]
 
 
 def _plot_topo(info=None, times=None, show_func=None, layout=None,
-               decim=None, vmin=None, vmax=None, ylim=None, colorbar=None,
+               vmin=None, vmax=None, ylim=None, colorbar=None,
                border='none', axis_facecolor='k', fig_facecolor='k',
                cmap='RdBu_r', layout_scale=None, title=None, x_label=None,
-               y_label=None, vline=None, font_color='w'):
+               y_label=None, vline=None, xticks=2, yticks=None,
+               font_color=None, linewidth=1, internal_legend=False,
+               external_legend=False, fontsize=None,
+               ylim_dict=None, show_names=None, plot_type=None,
+               external_scale=1):
     """Helper function to plot on sensor layout"""
     import matplotlib.pyplot as plt
 
     # prepare callbacks
     tmin, tmax = times[[0, -1]]
     on_pick = partial(show_func, tmin=tmin, tmax=tmax, vmin=vmin,
-                      vmax=vmax, ylim=ylim, x_label=x_label,
-                      y_label=y_label, colorbar=colorbar)
+                      vmax=vmax, ylim=ylim, colorbar=colorbar)
+
+    if xticks == 2:
+        xticks = [0.0, (tmin if np.abs(tmin) > np.abs(tmax) else tmax) / 2]
+        xticks = [round(.05 * round(float(x) / .05), 2) for x in xticks]
+    elif isinstance(xticks, int):
+        endpoints = times[0], times[-1]
+        xticks = [round(.05 * round(float(x) / .05), 2)
+                  for x in np.linspace(*endpoints, num=xticks + 2)][1:-1]
```

Comment:
is this somehow a 'smart' ticks generator? All these are great @jona-sassenhagen but I'd vote for breaking it into smaller PRs with more documentation and comments.